src/main/java/jeff/parser/Parser.java

```diff
@@ -0,0 +1,114 @@
+package jeff.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import jeff.command.AddCommand;
+import jeff.command.Command;
+import jeff.command.DateCommand;
+import jeff.command.DeleteCommand;
+import jeff.command.ExitCommand;
+import jeff.command.FindCommand;
+import jeff.command.ListCommand;
+import jeff.command.MarkCommand;
+import jeff.command.UnmarkCommand;
+import jeff.exception.JeffException;
+import jeff.task.Task;
+
+/**
+ * Represents a filter to sort out the user's input.
+ */
+public class Parser {
+    /**
+     * Returns a Command based on the user's input.
+     *
+     * @param input User's input.
+     * @return a Command.
+     * @throws JeffException if the user's input is not a command available.
+     */
+    public static Command parse(String input) throws JeffException {
+        // Filters the input based on the command given
+        if (input.equals("list")) {
+            return new ListCommand(input);
+        } else if (input.equals("bye")) {
+            return new ExitCommand(input);
+        } else if (input.startsWith("mark")) {
+            return new MarkCommand(input);
+        } else if (input.startsWith("unmark")) {
+            return new UnmarkCommand(input);
+        } else if (input.startsWith("delete")) {
+            return new DeleteCommand(input);
+        } else if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+            return new AddCommand(input);
+        } else if (input.startsWith("task")) {
+            return new DateCommand(input);
+        } else if (input.startsWith("find")) {
+            return new FindCommand(input);
+        } else {
+            throw new JeffException();
+        }
```

Comment:
Should the logic of parsing the first word in the string be extracted out?

You can consider creating an Enum and letting said Enum handle the logic of parsing the first word into a relevant command. This way, the chain of if-else statements can be replaced with a switch statement for better readability.

Reply:
Thanks for pointing it out!

I wanted to prompt the user with the format of the command when they forget to put a spacing after the command word. For example, "findborrow book". If i try to use an Enum and extract the first word, the first word in this case would be "findborrow" instead of "find". This would then prompt a different message instead and thus I didn't want to use Enum for this.

If you have a solution to this, do let me know!